new_reduce.rb

```diff
@@ -0,0 +1,9 @@
+class Array
+  def my_reduce(joint)
+    if block_given?
+      self.each do |i|
+        # something clever
```

Comment:
@bion I'm totally lost on this one.  Are you supposed to be able to individually access the params that you pass with the block, i.e. `acc` and `element` from `a.my_reduce { | acc, element | acc += element }`?